src/solvers/prop/bdd_expr.cpp

```diff
@@ -91,6 +91,30 @@ bddt bdd_exprt::from_expr(const exprt &expr)
   return from_expr_rec(expr);
 }
 
+/// Conjunction of two expressions. If the second is already an `and_exprt`
+/// adds to its operands instead of creating a new expression.
```

Comment:
The left hand side is expected to come from bdd variable at all the call sites, so it can't be a conjunction. We save one if condition at each node that way.